MapboxNavigation/RouteMapViewController.swift

```diff
@@ -496,11 +496,7 @@ extension RouteMapViewController: MGLMapViewDelegate {
     }
 
     func updateShield(for controller: RouteManeuverViewController) {
-        let currentLegProgress = routeController.routeProgress.currentLegProgress
-
-        guard let upComingStep = currentLegProgress?.upComingStep else { return }
-        guard let ref = upComingStep.codes?.first else { return }
-        guard controller.shieldImage == nil else { return }
```

Comment:
Keeping this guard would act as a failsafe for caching.

Currently, we also call `updateShield(for:)` from `notifyDidChange` one time per second. Can we try to limit that to one request per `willTransitionTo`?